actionview/lib/action_view/test_case.rb

```diff
@@ -39,6 +39,84 @@ def initialize
       end
     end
 
+    module Decoding
+      extend ActiveSupport::Concern
+
+      class_methods do
+        def inherited(descendant)
+          super
+
+          descendant.class_attribute :decoders, default: decoders.dup
+        end
+
+        # Register a callable to decode rendered content for a given MIME type
+        #
+        # Each registered decoder will also define a <tt>#rendered_$MIME</tt> helper
+        # method, where <tt>$MIME</tt> corresponds to the value of the
+        # <tt>mime</tt> argument
+        #
+        # === Arguments
+        #
+        # <tt>mime</tt> - Symbol the MIME Type name for the rendered content
+        # <tt>callable</tt> - Callable to decode the String. Accepts the String
+        #                     value as its only argument
+        # <tt>block</tt> - Block serves as the decoder when the
+        #                  <tt>callable</tt> is omitted
+        #
+        # By default, ActionView::TestCase defines a decoder for:
+        #
+        #   * :html - returns an instance of Nokogiri::XML::Node
+        #   * :json - returns an instance of ActiveSupport::HashWithIndifferentAccess
+        #
+        # Each pre-registered decoder also defines a corresponding helper:
+        #
+        #   * :html - defines `rendered_html`
+        #   * :json - defines `rendered_json`
+        #
+        # === Examples
+        #
+        # To parse the rendered content into RSS, register a call to <tt>RSS::Parser.parse</tt>:
+        #
+        #     register_decoder :rss, -> rendered { RSS::Parser.parse(rendered) }
+        #
+        #     test "renders RSS" do
+        #       article = Article.create!(title: "Hello, world")
+        #
+        #       render formats: :rss, partial: article
+        #
+        #       assert_equal "Hello, world", rendered_rss.items.last.title
+        #     end
+        #
+        # To parse the rendered content into a Capybara::Simple::Node,
+        # re-register an <tt>:html</tt> decoder with a call to
+        # <tt>Capybara.string</tt>:
+        #
+        #     register_decoder :html, -> rendered { Capybara.string(rendered) }
+        #
+        #     test "renders HTML" do
+        #       article = Article.create!(title: "Hello, world")
+        #
+        #       render partial: article
+        #
+        #       rendered_html.assert_css "h1", text: "Hello, world"
+        #     end
+        def register_decoder(mime, callable = nil, &block)
```

Comment:
I chose `decoder` here, since it isn't "encoding" the value the way that the [ActionDispatch::Testing::RequestEncoder](https://github.com/rails/rails/blob/44bd6e7acaa91dd59c260beea6863a49b84f6039/actionpack/lib/action_dispatch/testing/request_encoder.rb) does. Would `register_parser` be more self-documenting?

Reply:
Or maybe `register_renderer` 🤔

Reply:
I think `register_parser` or `register_renderer` is clearer as well.